File: src/main/java/com/jesperancinha/service/containers/CountryContainer.java
package com.jesperancinha.service.containers;

import com.jesperancinha.service.pojos.Country;
import lombok.Getter;

/**
 * Created by joaofilipesabinoesperancinha on 01-08-16.
 */
@Getter
public class CountryContainer {

    private Object[] FILE_HEADER = new Object[]{
            "id",
            "code",
            "name",
            "continent",
            "wikipedia_link",
            "keywords"
    };

    private Country country;

    public CountryContainer() {

    }
}

File: airports-app/src/main/kotlin/org/jesperancinha/airports/containers/CountryContainer.kt
package org.jesperancinha.airports.containers

import com.fasterxml.jackson.dataformat.csv.CsvMapper
import com.fasterxml.jackson.dataformat.csv.CsvSchema
import org.jesperancinha.airports.pojos.Country
import java.io.InputStreamReader

class CountryContainer {
    private val mapper by lazy { CsvMapper() }
    private val schema by lazy {
        CsvSchema.builder()
            .setColumnSeparator(',')
            .addColumn("id")
            .addColumn("code")
            .addColumn("name")
            .addColumn("continent")
            .addColumn("wikipediaLink")
            .addColumn("keywords")
            .build()
    }
    
    val countries by lazy {
        javaClass.getResourceAsStream("/countries.csv")?.use { stream ->
            mapper
                .reader(Country::class.java)
                .with(schema)
                .readValues<Country>(InputStreamReader(stream))
                .readAll()
        } ?: emptyList()
    }
}